src/plugins/telemetry_collection_manager/server/plugin.ts

```diff
@@ -227,15 +227,21 @@ export class TelemetryCollectionManagerPlugin
       if (statsCollectionConfig) {
         try {
           const usageData = await this.getUsageForCollection(collection, statsCollectionConfig);
-          if (usageData.length) {
+          const optedInUsageData = usageData.filter(isClusterOptedIn);
```

Comment:
NOTE: This means that opted-out clusters won't show any telemetry sample in the flyout when requested. Does it make sense? Should we add a warning in the UI explaining this?

Reply:
Actually, it _could_ happen that we show _monitored_ clusters but not the local telemetry 😕

Reply:
Since the filter was originally intended to filter monitoring-sourced telemetry clusters that are opted-in/out. I'm moving the filter to that place only.